### PR TITLE
Tighten updateable columns of sync classify step

### PR DESCRIPTION
**Pull request by calherries**
This PR tightens the `updateable-columns` so that the updateable column set depends on the model. This is mostly for documentation purposes.

### CODE REVIEW

File: src/metabase/sync/analyze/classify.clj [hunk 1]
```diff
@@ -59,15 +62,12 @@
...
         (throw (Exception. (format "Classifiers are not allowed to set the value of %s." k)))))
     ;; cool, now we should be ok to update the model
     (when values-to-set
-      (t2/update! (if (mi/instance-of? :model/Field original-model)
-                    :model/Field
-                    :model/Table)
-          (u/the-id original-model)
-        values-to-set)
+      (t2/update! (mi/model original-model) (u/the-id original-model)
+                  values-to-set)
```
Comment (calherries) [inline src/metabase/sync/analyze/classify.clj:70]: This is unrelated. Just nice to remove code

File: src/metabase/sync/analyze/classify.clj [hunk 2]
```diff
@@ -38,9 +38,12 @@
...
 
-(def ^:private values-that-can-be-set
-  "Columns of Field that classifiers are allowed to set."
-  #{:semantic_type :preview_display :has_field_values :entity_type})
+(defn- updateable-columns
+  "Columns of Field or Table that classifiers are allowed to be set."
+  [model]
+  (case model
+   :model/Field #{:semantic_type :preview_display :has_field_values}
+   :model/Table #{:entity_type}))
```
[the remark this answers is not in the document]
Comment (piranha) [reply src/metabase/sync/analyze/classify.clj:46]: Change looks reasonable, but what's the reason for `case` instead of just a map?

[the remark this answers is not in the document]
Comment (calherries) [reply src/metabase/sync/analyze/classify.clj:46]: I have no preference either way. What's the benefit of a map you see?

[the remark this answers is not in the document]
Comment (crisptrutski) [reply src/metabase/sync/analyze/classify.clj:46]: The case is nice to avoid debugging stray nils. I'd only use a map here if you'd intentionally want to have that fallback, since it's more code and also less efficient.

**Review by crisptrutski — APPROVED** [review]

File: src/metabase/sync/analyze/classify.clj (hunk 2 above)
[the remark this answers is not in the document]
Comment (piranha) [reply src/metabase/sync/analyze/classify.clj:46]: Well, that it kind of seems like configuration rather than execution, which is a map usually :)

[the remark this answers is not in the document]
Comment (piranha) [reply src/metabase/sync/analyze/classify.clj:46]: Also, "less efficient" is funny given that you don't need to execute map and construct those sets every time this function is executed.

[the remark this answers is not in the document]
Comment (calherries) [reply src/metabase/sync/analyze/classify.clj:46]: `case` is actually slightly faster than using a `map`.

```clojure
(defn- updateable-columns
  [model]
  (case model
   :model/Field #{:semantic_type :preview_display :has_field_values}
   :model/Table #{:entity_type}))

(def updateable-columns-2
  {:model/Field #{:semantic_type :preview_display :has_field_values}
   :model/Table #{:entity_type}})

(time (dotimes [_ 100000000]
        (contains? (updateable-columns :model/Table) :entity_type)))
"Elapsed time: 1712.516917 msecs"

(time (dotimes [_ 100000000]
        (contains? (updateable-columns-2 :model/Table) :entity_type)))
"Elapsed time: 1858.564167 msecs" 

(time (dotimes [_ 100000000]
        (contains? (get updateable-columns-2 :model/Table) :entity_type)))
"Elapsed time: 1949.7645 msecs"
```

don't ask me why though

[the remark this answers is not in the document]
Comment (piranha) [reply src/metabase/sync/analyze/classify.clj:46]: wtf 🤣

[the remark this answers is not in the document]
Comment (crisptrutski) [reply src/metabase/sync/analyze/classify.clj:46]: I hadn't thought about the allocations, and shocked to somehow still be right 😂

I was only talking about the choice between mechanisms in the abstract. It's a pity the JIT can't deduce that they are effectively immutable and hoist it #haskellenvy

[the remark this answers is not in the document]
Comment (piranha) [reply src/metabase/sync/analyze/classify.clj:46]: > I was only talking about the choice between mechanisms in the abstract.

What about data over functions for configuration? 😁

[the remark this answers is not in the document]
Comment (crisptrutski) [reply src/metabase/sync/analyze/classify.clj:46]: Damn now you're bringing out the big guns 😂 Can't argue with that
